wagtail/core/blocks/base.py

```diff
@@ -542,6 +542,9 @@ def __init__(self, block=None, **kwargs):
     def clean(self, value):
         return self.block.clean(value)
 
+    def has_changed(self, initial_value, data_value):
```

Comment:
was that all it took to fix `form.has_changed` being True for StreamFields? 🤯

Reply:
With this change, https://github.com/wagtail/wagtail/blob/master/wagtail/admin/views/pages.py#L503 should also change to something along the lines of:

> if there are content changes then create a new revision, otherwise use `latest_revision`, regardless of wether or not this is a workflow action

I think that is, but it has been a long day so will revisit tomorrow